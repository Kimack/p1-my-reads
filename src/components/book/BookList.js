import React from 'react'
import Book from './Book.js'
import PropTypes from 'prop-types';

const BookList = ({ books, shelves, onBookShelfChange }) => (
    <ol className="books-grid">
        {
            books.map((book) => (
                <li key={book.id}>
                    <Book
                        book={book}
                        shelves={shelves}
                        onBookShelfChange={onBookShelfChange} />
                </li>
            ))
        }
    </ol>
)

BookList.propTypes = {
	/**
	 * Array of current books on any shelf
	 */
	books: PropTypes.array.isRequired,
	/**
	 * Available shelves array
	 */
    shelves: PropTypes.array.isRequired,
	/**
	 * Function that will be called when the shelf is changed
	 */
	onBookShelfChange: PropTypes.func.isRequired
}

export default BookList;