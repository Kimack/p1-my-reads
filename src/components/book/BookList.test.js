import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import BookList from './BookList';

configure({ adapter: new Adapter() });

describe('BookList instance is rendered corretly', () => {
    it('renders without crashing', () => {
        const component = <BookList books={[{id:0}, {id:1}]} shelves={[]} onBookShelfChange={() => { }} />;
        expect(shallow(component)).toMatchSnapshot();
    });
});