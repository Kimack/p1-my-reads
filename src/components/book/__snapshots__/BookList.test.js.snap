// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList instance is rendered corretly renders without crashing 1`] = `
<ol
  className="books-grid"
>
  <li
    key="0"
  >
    <Book
      book={
        Object {
          "id": 0,
        }
      }
      onBookShelfChange={[Function]}
      shelves={Array []}
    />
  </li>
  <li
    key="1"
  >
    <Book
      book={
        Object {
          "id": 1,
        }
      }
      onBookShelfChange={[Function]}
      shelves={Array []}
    />
  </li>
</ol>
`;
